### DISCUSSION TITLE
Strange behaviour with `delete-selection-mode`

### DISCUSSION THREAD

**Discussion by Mango0x45** [minad/corfu#506] [Q&A]
I am using the builtin `delete-selection-mode` in Emacs, so when I start typing with an active region I expect it to get replaced.  Usually this works just fine:

```
# Brackets denote active region

foo [bar baz]
foo 123  # Typed ‘123’
```

For whatever reason though, when `corfu-mode` is active, the cursor doesn’t actually move after typing the first character that deletes the selection so I actually get the following undesired behaviour:

```
foo [bar baz]
foo 231  # Typed ‘123’, but point was never moved beyond ‘1’
```

This is an issue I’ve had only with `corfu-mode`, and I would love to know how/if it’s possible to fix this.

**Comment by minad** [comment]
Hi, I couldn't reproduce this in a brief try.

**Reply by Mango0x45** [reply]
So I investigated it some more and the issue seemingly solved itself until today, when I realized what was going on.  It seems that this issue exists specifically if I am deleting a selected region inside of a comment or string.

Very interestingly though, this does _not_ occur inside of single quotes in comments:

```elisp
;; Deleting this text causes issues `but not this text!', isn’t that weird?
```

I am not at all familiar with how Corfu functions, but it seems like this issue is happening in places where Corfu isn’t meant to be showing me completions?

EDIT: The issue seemingly solves itself when I remove `corfu--auto-post-command` from `post-command-hook`.

**Reply by Mango0x45** [reply]
Ok so… I’m getting this in other places now too where corfu _should_ be giving me completion. Most recently with a struct member name in `c-ts-mode`[1].  It’s pretty random but definitely common, and always goes away the moment I disable Corfu.

[1]: The exact example was:

```c
match_t m = {
    .hl /* <-- This name */ = ...;
};
```
